consensus-types/interfaces/beacon_block.go

```diff
@@ -125,8 +132,12 @@ type ExecutionData interface {
 	WithdrawalsRoot() ([]byte, error)
 	ValueInWei() (math.Wei, error)
 	ValueInGwei() (uint64, error)
-	DepositReceipts() ([]*enginev1.DepositReceipt, error)
-	WithdrawalRequests() ([]*enginev1.ExecutionLayerWithdrawalRequest, error)
+}
+
+type ExecutionDataElectra interface {
```

Comment:
Shouldn't there be `ExecutionDataCapella` and `ExecutionDataDeneb` too? That will allow us to remove unnecessary methods from previous forks.

Reply:
This is an incremental step, but yes I would like us to add interfaces for each fork.